pkg/gitrepo/error.go

```diff
@@ -40,3 +40,12 @@ var referenceNotFoundError = &microerror.Error{
 func IsReferenceNotFound(err error) bool {
 	return microerror.Cause(err) == referenceNotFoundError
 }
+
+var repositoryNotFoundError = &microerror.Error{
+	Kind: "repositoryNotFoundError",
+}
+
+// IsRepositoryNotFound asserts referenceNotFoundError.
+func IsRepositoryNotFound(err error) bool {
```

Comment:
Do we need that for something?

Reply:
Don't we always define an `IsX(err error) bool` function for every error `X`?

Reply:
I guess more idiomatic would be to create an internal error matcher, but we would still keep this function according to `fmt`. See https://github.com/giantswarm/fmt/blob/master/go/errors.md#handling-third-party-library-errors

Reply:
> Don't we always define an `IsX(err error) bool` function for every error `X`?

Yes we do, but only for errors that need to be handled. IMO this shouldn't be done "just in case" because it's extending the public API of the library and potential breaking changes in the dependency may make it harder to maintain the contract.

For the ideas of the microerror future we can discuss here https://github.com/giantswarm/giantswarm/issues/7139

Reply:
Yes, I will be using it in [opsctl](https://github.com/giantswarm/opsctl/blob/d9deef6d6e966624819c4d03f2818eb64ac14189/pkg/cmd/release/deployer.go#L260).

Reply:
Let's please add changelog and LGTM then.